web/modules/custom/joinup_migrate/joinup_migrate.post_update.php

```diff
@@ -159,3 +158,86 @@ function joinup_migrate_post_update_more_redirects() {
     }
   }
 }
+
+/**
+ * Fix distribution files with the same name [ISAICP-3968].
+ */
+function joinup_migrate_post_update_same_name_distribution_files() {
+  /** @var \Drupal\Core\File\FileSystemInterface $file_system */
+  $file_system = \Drupal::service('file_system');
+  /** @var \Drupal\Component\Uuid\UuidInterface $uuid_generator */
+  $uuid_generator = Drupal::service('uuid');
+  /** @var \Drupal\Core\Entity\ContentEntityStorageInterface $redirect_storage */
+  $redirect_storage = \Drupal::entityTypeManager()->getStorage('redirect');
+
+  $legacy_db = Database::getConnection('default', 'migrate');
+  $legacy_db_name = $legacy_db->getConnectionOptions()['database'];
+
+  /** @var \Drupal\Core\Database\Query\SelectInterface $query */
+  $query = Database::getConnection()->select('file_managed', 'f')
+    ->fields('f', ['fid', 'uri'])
+    ->fields('d6f', ['filename', 'filepath'])
+    ->orderBy('f.uri')
+    ->condition('f.uri', 'public://distribution/%', 'LIKE');
+  $query->join("$legacy_db_name.files", 'd6f', 'f.fid = d6f.fid');
+  $file_groups = [];
+  foreach ($query->execute()->fetchAll() as $row) {
+    $dir = substr($file_system->dirname($row->uri), 22);
+    // We accept only date formatted dirs.
+    if (preg_match('@^\d{4}\-\d{2}$@', $dir)) {
+      $file_groups[$dir][$row->filename][$row->fid] = $row->filepath;
+    }
+  }
+
+  // Keep only duplicates.
+  array_walk($file_groups, function (array &$file_group) {
+    $file_group = array_filter($file_group, function (array $files) {
+      return count($files) > 1;
+    });
+  });
+  $file_groups = array_filter($file_groups, function (array $file_group) {
```

Comment:
What is this needed for? Is there more to filter? I couldn't understand 😭 
Anyway, if you are trying to remove empty arrays, it's enough to call array_filter() alone.

Reply:
That was just for removing empty dirs resulted from the previous step. You're right, `array_filter()` is enough but the function was left there from a previous iteration when I tried to fix both steps in a single statement. However, I removed the line because it's useless.